front/components/assistant/AssistantDetails.tsx

```diff
@@ -239,76 +241,101 @@ export function AssistantDetails({
     actions,
   }: {
     actions: AgentConfigurationType["actions"];
-  }) =>
-    !!actions.length && (
-      <>
-        {actions.map((action, index) =>
-          isDustAppRunConfiguration(action) ? (
-            <div className="flex flex-col gap-2" key={`action-${index}`}>
-              <div className="text-lg font-bold text-element-800">Action</div>
-              <DustAppSection dustApp={action} owner={owner} />
-            </div>
-          ) : isRetrievalConfiguration(action) ? (
-            <div className="flex flex-col gap-2" key={`action-${index}`}>
-              <div className="text-lg font-bold text-element-800">
+  }) => {
+    const [retrievalAction, otherActions] = useMemo(() => {
+      return actions.reduce(
+        ([dataSources, otherActions], a) => {
+          if (isRetrievalConfiguration(a)) {
+            dataSources.push(a);
+          } else {
+            otherActions.push(a);
+          }
+          return [dataSources, otherActions];
+        },
+        [
+          [] as RetrievalConfigurationType[],
+          [] as AgentActionConfigurationType[],
+        ]
+      );
+    }, [actions]);
+
+    return (
+      !!actions.length && (
+        <>
+          {retrievalAction.length > 0 && (
```

Comment:
nit: code consistency with above
```suggestion
          {!!retrievalAction.length && (
```